### PR TITLE
override exclude_none default to True and sanitize subjects list

### PR DESCRIPTION
**Pull request by sblack-usu**

### CODE REVIEW

File: hsmodels/schemas/rdf/validators.py [hunk 1]
```diff
@@ -24,6 +24,15 @@ def rdf_parse_identifier(cls, value):
     return value
 
 
+def subjects_constraint(cls, subjects):
+    """Removes empty/None and duplicates"""
+    validated = []
+    for subject in subjects:
+        if subject and subject not in validated:
+            validated.append(subject)
+    return validated
```
Comment (hyi) [inline hsmodels/schemas/rdf/validators.py:33]: @sblack-usu it'd be better to trim each subject before line 31. Also, it is probably better to convert subjects into lower case to account for same subjects with only case difference? Just some minor edge cases for consideration, but not a big deal at all.

Comment (sblack-usu) [reply hsmodels/schemas/rdf/validators.py:33]: I can add the trim.

Comment (sblack-usu) [reply hsmodels/schemas/rdf/validators.py:33]: I don't think we should be lowercasing subjects, hydroshare does not do that today.  I think discover might do that when indexing.

Comment (hyi) [reply hsmodels/schemas/rdf/validators.py:35]: OK. Sounds good.

Comment (hyi) [reply hsmodels/schemas/rdf/validators.py:35]: This is really nitpicking, but it'd be better to strip() the subject before checking it on line 31. I am approving the PR now, feel free to merge it or make this small change then merge it.

**Review by hyi — APPROVED** [review]

File: hsmodels/schemas/rdf/validators.py (hunk 1 above)
Comment (sblack-usu) [reply hsmodels/schemas/rdf/validators.py:34]: It's a valid critique, done.

**Pull request merged**